### DISCUSSION TITLE
Index of element in list

### DISCUSSION THREAD

**Discussion by lilrosalyn** [SkriptLang/Skript#4478] [Suggestions] [answered]
Lists should be able to be searched for elements, so lists can be reliably manipulated.

**Example Usage**
```vb
set {_l::*} to 1, 2, 2, and 3
set {_i::*} to indices of 2 in {_l::*} # Returns [2, 3]
set {_i::*} to indices of 1 in {_l::*} # Returns [1]

set {_kvList::giga} to "chad"
set {_i::*} to indices of "chad" in {_kvList::*} # Returns ["giga"]
```

**Example use case**
```vb
# Let's say we have a list, {_chat::*}, which contains the last 100 chat messages sent. The index 100 is the newest chat message sent, and the index 0 is the oldest chat message sent, if any.
# If we know the chat message "hey guys!" is at index 54, we can just access that message using {_list::54}, but if somebody sends a chat message, then {_chat::54} would be a different chat message, for example, "icecream", instead of "hey guys!", like we expected.

# This problem could be solved using the "index of element in list" expression, using:
set {_c} to indices of "hey guys!" in {_chat::*}
```

**Comment by bluelhf** [comment]
This can technically already be done\*, but having specific syntax for it would be neat!

*`indices of {_list::*} where [{_list::%input%} is 2]`

**Reply by lilrosalyn** [reply]
Specific syntax could also be helpful because the best way to do it in Skript has O(n) access, instead of (ideally) O(1) access.

**Reply by bluelhf** [reply]
Even with specific syntax I don't think searching a map like that is possible in O(1) time without some kind of external map... How would you implement this in under O(n)?

**Comment by TPGamesNL** [comment, marked as answer]
See https://github.com/SkriptLang/Skript/issues/3933
